src/hotspot/share/oops/objArrayKlass.hpp

```diff
@@ -35,6 +35,8 @@ class ClassLoaderData;
 class ObjArrayKlass : public ArrayKlass {
   friend class VMStructs;
   friend class JVMCIVMStructs;
+  friend class oopFactory;
+  friend class Deoptimization;
```

Comment:
If you want you could consider sorting the friend declarations (here and in the other place where you added it)